waku/v2/peermanager/peer_connector.go

```diff
@@ -277,10 +276,9 @@ func (c *PeerConnectionStrategy) dialPeer(pi peer.AddrInfo, sem chan struct{}) {
 	ctx, cancel := context.WithTimeout(c.Context(), c.dialTimeout)
 	defer cancel()
 	err := c.host.Connect(ctx, pi)
-	if err != nil && !errors.Is(err, context.Canceled) {
-		c.addConnectionBackoff(pi.ID)
-		c.host.Peerstore().(wps.WakuPeerstore).AddConnFailure(pi.ID)
-		c.logger.Warn("connecting to peer", logging.HostID("peerID", pi.ID), zap.Error(err))
+	if err != nil {
+		c.pm.HandleDialError(err, pi.ID)
+		return
```

Comment:
returning from here means we should release semaphore which is done in line 284.

Reply:
I see, it seemed strange to record a dial error and then reset connection failures right after. I removed the return and use an if/else statement to only reset connection failures if there is no error